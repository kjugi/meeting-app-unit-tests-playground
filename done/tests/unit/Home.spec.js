import { createWrapper, createStore } from '../factory'
import HomePage from '@/views/Home.vue'

describe('Home page', () => {
  it('meeting list contains elements from store', () => {
    const localStore = createStore({ state: {
      meetingList: [
        {
          'date': '2020-03-27',
          'who': 'Example Person1'
        },
        {
          'date': '2020-03-27',
          'who': 'Example Person2'
        }
      ]
    }})
    const wrapper = createWrapper(HomePage, { store: localStore })

    expect(wrapper.findAll('.home__item')).toHaveLength(2)
    expect(wrapper).toMatchSnapshot()
  })

  it('render empty list message when don\'t have items', () => {
    const wrapper = createWrapper(HomePage)

    expect(wrapper.find('.home div').text()).toBe('Meeting list is empty!')
  })
})
