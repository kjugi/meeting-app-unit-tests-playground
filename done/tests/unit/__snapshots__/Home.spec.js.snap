// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Home.vue meeting list contains elements from store 1`] = `
<div class="home">
  <div class="home__list">
    <div class="home__item">
      <h2>
        Example Person1
      </h2> <span class="date">
        2020-03-27
      </span>
    </div>
    <div class="home__item">
      <h2>
        Example Person2
      </h2> <span class="date">
        2020-03-27
      </span>
    </div>
  </div>
</div>
`;
